### ISSUE TITLE
DictionarySource Performance

### ISSUE THREAD

**Issue by karljj1** [axuno/SmartFormat#186]
Hey.

I noticed that the DictionarySource is iterating through all the keys to find a match, this is probably because of the case sensitivity option. I wonder if it would be possible to use the TryGetValue method when the Dictionary is using the same EqualityComparer?

Something like this?

```
if (current is Dictionary<string, object> dict && dict.Comparer == selectorInfo.FormatDetails.Settings.GetCaseSensitivityComparer())
{
    if (dict.TryGetValue(selector, out var result))
    {
        selectorInfo.Result = result;
        return true;
    }
}
```

**Comment by axunonb** [comment]
Hi
That's right: it's because of case-sensitivity. The case-sensitivity setting of SmartFormat works, no matter how the dictionary instance is created. It could e.g. be ```new Dictionary<string, object>(20, StringComparer.OrdinalIgnoreCase)```.
We could of course say, that the comparison setting is fully up to the caller, and the SmartFormat setting will not apply.
Is that what you have in mind?
Did you already compare performance of ```TryGetValue``` against the current implementation?

**Comment by axunonb** [comment]
Resolved with  PR #189, part of [v3.0.0-alpha.1](https://github.com/axuno/SmartFormat/releases/tag/v3.0.0-alpha.1)
